tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/JSDesugaring.scala

```diff
@@ -1129,14 +1165,16 @@ private[emitter] class JSDesugaring(internalOptions: InternalOptions) {
          */
         case Match(selector, cases, default) =>
           unnest(selector) { (newSelector, env0) =>
-            implicit val env = env0
+            implicit val env = env0.withDefaultBreakTargets(tailPosLabels)
             extractLet { newLhs =>
               val newCases = {
                 for {
                   (values, body) <- cases
                   newValues = (values map transformExpr)
                   // add the break statement
-                  newBody = js.Block(pushLhsInto(newLhs, body), js.Break())
+                  newBody = js.Block(
+                      pushLhsInto(newLhs, body, tailPosLabels),
```

Comment:
Just to verify my understanding: You are allowed to pass `tailPosLabels` down here, since we manually add `js.Break()` at the end of this block, right?

Further: In some sense, adding the `tailPosLabels` to the default break targets is correct, but redundant, right?